game/atmosphere.rpy

```diff
@@ -49,142 +98,227 @@ init 0 python in jn_atmosphere:
+        day_clouds="clouds day heavy",
+        night_clouds="clouds night heavy",
     )
 
     WEATHER_RAIN = JNWeather(
         weather_type=JNWeatherTypes.rain,
-        day_image="sky_day rain",
+        day_sky="sky day rain",
+        night_sky="sky night rain",
         dim_image="dim medium",
+        day_clouds="clouds day heavy",
+        night_clouds="clouds night heavy",
         weather_sfx="mod_assets/sfx/rain_muffled.mp3"
     )
 
     WEATHER_THUNDER = JNWeather(
         weather_type=JNWeatherTypes.thunder,
-        day_image="sky_day thunder",
+        day_sky="sky day thunder",
+        night_sky="sky night thunder",
         dim_image="dim heavy",
+        day_clouds="clouds day thunder",
+        night_clouds="clouds night heavy",
         weather_sfx="mod_assets/sfx/rain_muffled.mp3"
     )
 
+    WEATHER_SNOW = JNWeather(
+        weather_type=JNWeatherTypes.snow,
+        day_sky="sky day snow",
+        night_sky="sky night overcast",
+        dim_image="dim light",
+        day_clouds="clouds day light",
+        night_clouds="clouds night light",
+    )
+
     WEATHER_SUNNY = JNWeather(
         weather_type=JNWeatherTypes.sunny,
-        day_image="sky_day sunny"
+        day_sky="sky day sunny",
+        night_sky="sky night sunny",
+        day_clouds="clouds day light",
+        night_clouds="clouds night light"
     )
 
     WEATHER_GLITCH = JNWeather(
         weather_type=JNWeatherTypes.glitch,
-        day_image="glitch_fuzzy")
-
-    __RANDOM_WEATHER_TYPES = [
-        WEATHER_OVERCAST,
-        WEATHER_RAIN,
-        WEATHER_THUNDER,
-        WEATHER_SUNNY
-    ]
+        day_sky="glitch_fuzzy",
+        night_sky="glitch fuzzy")
+
+    __WEATHER_CODE_REGEX_TYPE_MAP = {
+        ("^2[0-9][0-9]$"): WEATHER_THUNDER, # Thunder
+        ("^3[0-9][0-9]$"): WEATHER_RAIN, # Drizzle
+        ("^5[0-9][0-9]$"): WEATHER_RAIN, # Rain
+        ("^6[0-9][0-9]$"): WEATHER_SNOW, # Snow
+        ("^7[0-9][0-9]$"): WEATHER_OVERCAST, # Misc (mist, tornado, sandstorms, etc.)
+        ("(800|801|802)"): WEATHER_SUNNY, # Clear/light clouds
+        ("(803|804)"): WEATHER_OVERCAST, # Clouds
+    }
 
     current_weather = None
 
-    def show_current_sky(with_transition=True):
+    def update_sky(with_transition=True):
         """
         Shows the sky based on the sunrise/sunset times specified under the persistent.
 
         IN:
             with_transition - If True, will visually fade in the new weather
         """
-        if (store.jn_get_current_hour() > store.persistent.jn_sunrise_hour 
-            and store.jn_get_current_hour() <= store.persistent.jn_sunset_hour ):
-            if store.persistent.jn_random_weather:
-                show_random_sky(with_transition=with_transition)
-
-            elif not is_current_weather_sunny():
+        # API-based weather
+        if jn_preferences.weather.JNWeatherSettings(store.persistent.jn_weather_setting) == jn_preferences.weather.JNWeatherSettings.real_time:
```

Comment:
Is there any reason we re-initialize the enum here? Can't we just cast the latter half to int for the comparison?

Reply:
Good call, actually